vector/src/main/java/im/vector/app/features/settings/notifications/VectorSettingsNotificationPreferenceFragment.kt

```diff
@@ -98,14 +101,10 @@ class VectorSettingsNotificationPreferenceFragment @Inject constructor(
         findPreference<SwitchPreference>(VectorPreferences.SETTINGS_ENABLE_THIS_DEVICE_PREFERENCE_KEY)?.let {
             it.setTransactionalSwitchChangeListener(lifecycleScope) { isChecked ->
                 if (isChecked) {
-                    FcmHelper.getFcmToken(requireContext())?.let {
-                        pushManager.registerPusherWithFcmKey(it)
-                    }
+                    unifiedPushHelper.register(requireActivity())
```

Comment:
There is a (not new?) bug when playing with this settings. Background notification does not reappear when re-anabling notification for this session.
Also when the dialog is displayed and the user make a choice, it does not refresh the "Notification method" `summary`.

Reply:
The latter point if handled in b1e062a